### ISSUE TITLE
"Microsoft CodeAnalysis 2019" is older than NuGet version

### ISSUE THREAD

**Issue by kamyker** [MicrosoftDocs/visualstudio-docs#5433]
According to https://github.com/dotnet/roslyn-analyzers/issues/3616 VSIX extension will be deprecated in the future and currently is outdated. Docs miss that information.

Additionally: is there any method to make NuGet analyzers package work with software like Unity that generate solutions and make project properties unavailable?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: eee67c1d-5856-93e0-e2d3-e9551436aabc
* Version Independent ID: b756aa96-bbb1-40e8-34ca-97f9939bf7e9
* Content: [Install FxCop analyzers - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/install-fxcop-analyzers?view=vs-2019)
* Content Source: [docs/code-quality/install-fxcop-analyzers.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/install-fxcop-analyzers.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by WilliamAntonRohm** [comment]
@kamyker -- Kamil, thank you for your feedback and question. Here are a few options where you might consider asking your question:​  

- [https://social.msdn.microsoft.com/Forums/vstudio](https://social.msdn.microsoft.com/Forums/vstudio) 
- [https://www.stackoverflow.com](https://www.stackoverflow.com) 
- [https://www.reddit.com/r/VisualStudio/](https://www.reddit.com/r/VisualStudio) 

As well, you might also consider using the [Report a Problem](https://docs.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio) tool to report the issue. 

@Mikejo5000 -- Mike, please look into this issue.

**Comment by Mikejo5000** [comment]
@mavasani Is it correct yet for us to say that the VSIX is deprecated?

**Comment by mavasani** [comment]
@Mikejo5000 Yes, theoretically, the extension has very old CA rules (more then a year old) and we do not plan to support it further. In the very near future, these CA rules will be available in the .NET SDK itself, where users can turn them on by setting an MSBuild property on the project. Majority of them will be disabled/IDE suggestions by default to avoid noise, but users can easily enable/configure individual rule. This would allow users to easily enable the CA rules in their projects. We have also talked about allowing solution or VS level analyzer NuGet packages as a replacement for analyzer VSIX extensions, but that feature is not going to be implemented anytime soon.

To summarize, yes the extension is going away from the marketplace soon. NuGet package is currently our primary supported mechanism, and soon these analyzers with ship with the .NET SDK. I think we should set up a meeting to discuss the path forward and how/when to update the documentation.

**Comment by kamyker** [comment]
I've managed to get it working in Unity using https://github.com/tertle/com.bovinelabs.analyzers, could be added to docs.

**Comment by Mikejo5000** [comment]
Docs have been updated in regard to the VSIX and .NET5 changes, so we could probably close this.
Regarding third-party solution for Unity, @kamyker, does .NET5 eliminate the need to use the third-party package for Unity?

**Comment by kamyker** [comment]
NET5 is not supported in Unity, I've switched to Rider so issue not related to me anymore.

**Comment by WilliamAntonRohm** [comment]
@kamyker -- Kamil, thank you for clarifying.

I am closing this issue now.  You are welcome to @ mention me for any followup.
We hope to hear from you again.
